src/platform/common/utils/multiStepInput.ts

```diff
@@ -209,19 +208,19 @@ export class MultiStepInput<S> implements IMultiStepInput<S> {
                         const inputValue = input.value;
                         input.enabled = false;
                         input.busy = true;
-                        if (!(await validate(inputValue))) {
+                        const validationMessage = await validate(inputValue);
+                        if (!validationMessage) {
+                            input.validationMessage = '';
                             resolve(inputValue);
+                        } else {
+                            input.validationMessage = validationMessage;
                         }
                         input.enabled = true;
                         input.busy = false;
                     }),
-                    input.onDidChangeValue(async (text) => {
```

Comment:
I think we should bring this back, this will impact other places where this is used,
also i don't think it matters if we have this validation as the user types the text

Reply:
> vent doesn't catch the error. Maybe because validate takes longer now.

Even if thats the case, we'd still have the valdiation upon hitting enter.
& even if the values change, then those results would trump the previous error messages

Reply:
Validation on typing doesn't work because if the validation is the same error, it looks like it didn't do anything. It looks broken.

Reply:
Does that matter, if the code works as expected, then the user keeps seeing an error message
I don't see why they'd expect different errors when the underlying problem is still the same - i.e. url is invalid.

Reply:
Cause it looks like we didn't check again. They typed in something and got no changes whatsoever (even when the url isn't complete).

They're done when they hit enter. That's when I would expect it to validate.